### PR TITLE
fix(cli): fix lb4 discover --config missing provided options

### LINKED ISSUE #8963: lb4 datasource with -c option creates datasource file with errors
### Describe the bug

Creating a datasource with -c option and providing JSON data creates a datasource with errors.

Looks like when a datasource is being created with -c option, the `env` & a few other properties from yeoman are being added to the datasource configuration. The `env` property increases the number of lines in the datasource file to ~30K.

`.env` property being added to `this.options` also introduces the error of TypeError: Converting circular structure to JSON while discovering models by passing -c option.

### Logs

```shell
The logs show no errors.
```

### Additional information

You can see a sample datasource I created at my local system [here](https://github.com/aaqilniz/lb4-datasource-bug/blob/17dfb3036820d6d1c35270e263165f0d85f083e2/src/datasources/localmodels.datasource.ts#L14).

Adding a delete statement to remove `env` variable from `this.options` in the `setOptions` before it terminates resolves both problems.

I am attaching a screenshot too. 

image

### Reproduction

https://github.com/aaqilniz/lb4-datasource-bug

### LINKED ISSUE #8966: lb4 discover command with -c option fails to use those properties.
### Describe the bug

Running lb4 discover command with the --config/-c option doesn't recognize the properties. 

The `dataSource` field is being recognized but other fields like `all` & `schema` are not being recognized.

### Logs

_No response_

### Additional information

I have shared a sample repo for another issue I created that is somewhat relevant to this. The same repo could be used to reproduce this issue. 

### Reproduction

https://github.com/loopbackio/loopback-next/issues/8963

### PR DESCRIPTION
**Pull request by aaqilniz**
This PR fixes the issues lb4 discover & lb4 datasource ran with the --config option.

"fix #8963", "fix #8966".

Signed-off-by: Muhammad Aaqil <aaqilniz@yahoo.com>

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

### CODE REVIEW

**Comment by samarpanB** [comment]
@aaqilniz is this still work in progress ? I only see package lock files. If it is work in progress, can you change this to draft ? If it only requires dependencies upgrade, you may not need to raise a PR. That will automatically be handled by renovate bot. Please confirm.

**Comment by aaqilniz** [comment]
Hi @samarpanB. Thanks for the comment. The changes I actually made were in the file packages/cli/lib/base-generator.js. [Here](https://github.com/loopbackio/loopback-next/pull/9014/files#diff-49fcccd31a50c95987802f46120769203478d1efac4a9dfe7b81b8baf191b46f) is the link to the file.

While I was trying to run the tests at my local setup, the package-lock.json file got updated & pushed to the PR. Should I create another PR mentioning the changed file only?

File: packages/cli/lib/base-generator.js [hunk 1]
```diff
@@ -209,6 +209,15 @@ module.exports = class BaseGenerator extends Generator {
         `Package manager '${packageManager}' is not supported. Use ${supported}.`,
       );
     }
+
+    //In case lb4 is run with discover/datasource model, removing unwanted .env
+    if (
```
Comment (samarpanB) [inline packages/cli/lib/base-generator.js:214]: Its better to have this logic in individual generator classes. That will keep this base generator independent of child implementations. This function is overridden there already. Example here - https://github.com/loopbackio/loopback-next/blob/e854c2bc9924331fbb8882a9a87058335cc82939/packages/cli/generators/discover/index.js#L81

**Review by samarpanB — CHANGES_REQUESTED** [review]

File: packages/cli/lib/base-generator.js (hunk 1 above)
Comment (aaqilniz) [reply packages/cli/lib/base-generator.js:214]: You are right. Adding it to the relevant generator is better. I'll move the change to the discover generator. 
The confusion or question I had is that why this property .env is being added to the options object. Probably needed at some point but what would be a good place to remove it from the options object?

**Review by samarpanB — APPROVED** [review]

**Comment by aaqilniz** [comment]
@achrinza Can you please review this PR?

**Review by achrinza — APPROVED** [review]
We'll need to eventually find a "proper" fix for `env` and document its use and impact (i.e apply a genera fix across all generators)

However since the new changes are limited to specific generators, I think we should be good to land this PR.

**Comment by aaqilniz** [comment]
@achrinza You are right. There should be a proper way to handle the `env` but I couldn't figure out much as I am not that much familiar with the codebase. This was a quick fix I got to resolve the issue. 

All checks have passed so I think the PR can be merged, right?

**Comment by achrinza** [comment]
Thanks for your contributions, @aaqilniz! Your changes have been merged 🎉

**Comment by aaqilniz** [comment]
Awesome. Thanks. :)

**Pull request merged**